### DISCUSSION TITLE
Mode de tri comme choix a l'entrée d'un menu

### DISCUSSION THREAD

**Discussion by alKODIque** [Kodi-vStream/venom-xbmc-addons#3461] [Idées]
Bonjour,
Je me demandais s'il ne serait pas possible, depuis un Menu a l'intérieur d'une source, de pouvoir avoir une étape de tri AVANT d'arriver aux résultats?

Je m'explique..
Disons que je suis dans la source Pastebin, que j'entre dans le menu ''Films (nouveautés)''.  Au lieu d'obtenir directement les pochettes (résultats), obtenir plutot un choix de tri (exemple: nouveautés par notation, nouveautés par date, etc...) et lors de la sélection de l'ordre de tri voulue,, là, à ce moment recevoir les résultats....

J'ai pris cette idée depuis l'addon TMDb Helper. 
L'addon propose cette option lors de la création d'un widget depuis une liste Trakt (et seulement pour la création de widget).. Je me demande donc si ceci serait possible a réaliser depuis un menu de source vStream :P 

Voici une impression d'écran qui fait voir l'étape de Tri que j'essaie d'expliquer:
 
![Screenshot_20220204-174053](https://user-images.githubusercontent.com/49578130/152613033-dd8027aa-6772-49f2-bf4d-6b1429e16945.png)

(J'ai sélectionné: Listes Populaires / Nom de la liste /  -->  et j'obtiens les choix indiqués sur l'image avant d'obtenir le contenu de la liste)

**Comment by sizanic** [comment]
Bonjour,

L'exemple que tu prends avec Pastebin est plus qu'un exemple, car il n'y a qu'avec cette source que ce serait possible.
Pastebin connaît à l'avance l'ensemble des films, et peut appliquer un tri.

Les autres sources ne proposent qu'une page à la fois (chaque page du site associé), donc le tri sur l'ensemble des médias n'est pas possible.

Quant à ajouter une fenêtre supplémentaire systématique, cela est contraignant si on ne l'utilise jamais (si le tri par défaut nous satisfait).

Et comme on ne peut le faire qu'avec Pastebin, dans quel menu cela serait-il vraiment intéressant ?
Les menus proposés implique déjà un tri implicite.
(il n'y pas de menu ''Films (nouveautés)'' 🤔 )


Et dernière grosse contrainte technique, pour un tri par Note ou par Date, il faudrait connaître la note et la date de chaque film.
Et pour cela, avant de faire le tri, il faudrait récupérer les métadonnées de l'ENSEMBLE des médias 😱

**Reply by alKODIque** [reply]
... Mais je comprend bien la contrainte de récupérer toutes les métadonnées.... puisque par défaut, seulement 25 éléments a la fois sont récupérés :(

**Comment by alKODIque** [comment]
Dans les Genres!!! ce serait plus qu'utile , ce serait nécessaire hihi!! (minimalement, un tri par date --> voir num TMDb)

(Pour menu Nouveautés... je veux dire: nouveaux ajouts)

**Reply by sizanic** [reply]
Déjà on pourrait ajouter un vrai menu "Nouveautés" en utilisant justement ce num TMDB (on n'a pas les notations, mais on a bien tous les num TMDB).


Et comme "Notations" et "Date de sortie" ne peuvent pas être utilisés comme tri, il ne ne reste pas assez de possibilités pour mettre en place un choix de tri.

Mais on peut revoir les tris par défaut des menus disponibles.

Pour l'instant, l'ordre dépend de la catégorie :
- Dans (derniers ajouts), les films sont dans l'ordre des ajouts. -> Le tri est dans le titre 🤔 
- Dans (Ordre alphabétique), les films sont dans l'ordre alphabétique.  -> Le tri est dans le titre 🤔 
- Dans (Saga), les films sont dans l'ordre chronologique. -> Les films se suivent forcement par date 🤔 
- Dans (Liste), (Par Acteurs), (Par Réalisateurs), les films sont dans l'ordre chronologique inverse.  -> une filmographie se suit par date
- Dans (Genre) et (Résolution), les films sont dans l'ordre des ajouts. 
- Dans (Diffuseur), les séries sont dans l'ordre des ajouts.


Il n'y a vraiment que par (Genre), (Résolution) et (Diffuseur) que l'ordre pourrait être changé par un autre 🤔 
(et peut-être (Liste) aussi)

Et si on met par numTMDB pour ces menus ou pour seulement certains ?

**Reply by alKODIque** [reply]
Oui, je connais bien l'ordre de tri dans les menus... et malheureusement, je trouve que la catégorie ''nouveaux ajouts'' est inutilisable... la majorité des gens recherchent les nouveautés, les films récents... 

et le menu Saga: pareil. il n'y a aucune facon de trier autrewment. ex: un tri par genre serait génial dans les sagas... par titre c'est très long chercher ce que l'on veut.

Un tri par genre (dans les menus ou c'est piosssible) serait bien.
Un tri par nouveautés (ou num TMDb) serait plus que génial dans le menu par genre...
Le tri alpha est très bien comme ca: on sait ce que l'on veut lorsqu'on entre dans ce menu, c'est évident. (Dailleurs, il fonctionne bien même si tous les médias ne sont pas pré-scanné???)

S'il est impossible d'inclure des mode de tri du a la contrainte de l'extraction des métadonnées: est-il possible alors de créer de nouveaux menus? dans la mentalité du menu par ordre Alpha?

Exemble:
Menu -->Par genre
Sous-menu 1: --> par année
Sous-menu 2: --> alpha
Sous-menu3: --> par notation
Sous-menu 4 --> par date d'ajout
etc....

**Reply by sizanic** [reply]
> malheureusement, je trouve que la catégorie ''nouveaux ajouts'' est inutilisable... la majorité des gens recherchent les nouveautés, les films récents...

Je propose en complément un nouveau menu "Nouveautés", classé par idTMDB (pas besoin de toutes les métadonnées)


>  le menu Saga: pareil. il n'y a aucune facon de trier autrement. ex: un tri par genre serait génial dans les sagas...

Pas de trie par genre sans l'ensemble des métadonnées 😢 

> Un tri par genre (dans les menus ou c'est posssible) serait bien.

Dans aucun menu sans métadonnées.

> Un tri par nouveautés (ou num TMDb) serait plus que génial dans le menu par genre...

Enfin quelque chose qu'on peut faire 👍🏻 
Mais ce serait pour tout le temps, il remplace le tri en place (par date d'ajout).


> Le tri alpha est très bien comme ca: on sait ce que l'on veut lorsqu'on entre dans ce menu, c'est évident. (Dailleurs, il fonctionne bien même si tous les médias ne sont pas pré-scanné???)

Oui, le titre n'est pas une métadonnée à récupérer, on l'a déjà

**Reply by alKODIque** [reply]
ok je vois...
mais déja, s'il serait possible de trier les genres par numéro ID de TMDb... ce serait bien.  Et selon moi, serait ok pour ''tout le temps'', on a déja un menu ''nouveaux ajouts'', meme s'il n'est pas par genre. La quantité ne justifie pas que l'on tri les genre par derniers ajouts je crois...
 c'est mon avis mais on peut faire un sondage sur fb aussi :)

**Reply by sizanic** [reply]
Je propose de trier par ID les menus : 
- par genre
- par année (avec l'ID, ce serait un équivalent de date de sortie finalement)
- nouveautés (nouveau menu)

Il faut garder "derniers ajouts" pour les films sortis en salles un peu plus tot mais qu'on a mis du temps à récupérer, il s'est fait devancer, et il n'apparaitrait que sur les pages suivantes si on trie par ID.

**Reply by sizanic** [reply]
C'est traité pour les tris par ID des sous-catégories Genre et Années 👍🏻

**Comment by sizanic** [comment]
> est-il possible alors de créer de nouveaux menus ?

Exemble:
Menu -->Par genre
    Sous-menu 1: --> par année           -> **OUI**
    Sous-menu 2: --> alpha                  -> **OUI**
    Sous-menu3: --> par notation        -> **NON**
    Sous-menu 4 --> par date d'ajout  -> **OUI**
    etc....


Lorsqu'on navigue à l'aveugle dans le menu Genres, c'est pour chercher un film d'un certain genre sans considérer une année en particulier.

Mais je comprend que l'on puisse être dans le menu des Films de l'année 2021, à la recherche d'un film d'action.
=> Il y a bien ici deux filtres  (genre = Action, année = 2021), mais ce n'est pas deux tris.

Est-ce pas plutot un **filtre** multi-critères qui serait plus interessantes, au lieu d'un **tri** à l'interieur d'un **filtre**.

=> Ce qui permettrait également de créer des widgets.

Comme par exemple : 
- les series Netflix de 2021.
- les films d'action de 2022


=> Sauf que les critères sont assez réduits, on ne peut utiliser que : 
- Diffuseur de séries
- Genre
- Année

**Reply by alKODIque** [reply]
> > est-il possible alors de créer de nouveaux menus ?
> 
> Exemble: Menu -->Par genre Sous-menu 1: --> par année -> **OUI** Sous-menu 2: --> alpha -> **OUI** Sous-menu3: --> par notation -> **NON** Sous-menu 4 --> par date d'ajout -> **OUI** etc....
> 
> Lorsqu'on navigue à l'aveugle dans le menu Genres, c'est pour chercher un film d'un certain genre sans considérer une année en particulier.
> 

Je ne  suis pas dutout d'accord... Je navigue très souvent dans des genres qui m'intéressent d'avantage et je ne recherche pas dutout un titre en particulier... je veux voir quelles sont les nouveautés, les dernièeres sorties. Je ne veux pas dutout voir  en début de navigation un nouvel ajout d'un film de 1992... 

> Mais je comprend que l'on puisse être dans le menu des Films de l'année 2021, à la recherche d'un film d'action. => Il y a bien ici deux filtres (genre = Action, année = 2021), mais ce n'est pas deux tris.
> 
> Est-ce pas plutot un **filtre** multi-critères qui serait plus interessantes, au lieu d'un **tri** à l'interieur d'un **filtre**.
> 
> => Ce qui permettrait également de créer des widgets.
> 
> Comme par exemple :
> 
> * les series Netflix de 2021.
> * les films d'action de 2022
> 
> => Sauf que les critères sont assez réduits, on ne peut utiliser que :
> 
> * Diffuseur de séries
> * Genre
> * Année

Pour tous ces exemple... le meilleur exemple de ce qui serait au TOP est le mode de tri par défaut proposé par Kodi ;).... lol.. mais je comprend que ce n'est pas possible avec vStream / Pastebin

Médiathèque locale kodi: options de tri. Ici l'exemple est dans Genres / Action... mais chaque menu a ces options:

![Screenshot_20220204-191656](https://user-images.githubusercontent.com/49578130/152619935-70016306-a671-413a-ad5a-daeec8fd3186.png)

**Comment by alKODIque** [comment]
Mais tout ceci dit...
J'oubliais les modif a la dernière beta concernant la recherche... 🥇 
Avant aujourd'hui, j'avais (J'avoue) oublié et laissé pour compte la fonction car elle n'était pas très exhaustive...

La recherche étant maintenant beaucoup plus fonctionnelle, l'utiliser pourra compenser certains modes de tri ''par défaut'' que j'aurais aimé voir afin de compenser cette ancienne lacune 👍

**Comment by detobel36** [comment]
Personnellement, j'utilise TMDB pour faire la liste de film et puis "simplement" vStream pour rechercher et lancer le film.

Donc en fait on "inverse" la recherche des meta data. On commence par lister tous les films (par genre, année, note, ...). Et puis seulement quand on a choisi on utilise vStream pour rechercher le film et le lancer.
